### PR TITLE
Adds signup button tracking on pitch page

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?

Tracks everytime a user clicks signup on the pitch page
#### How should this be reviewed?

Enable GA debugging
Click signup on a pitch page
Verify event fired
#### Relevant tickets

https://trello.com/c/I6qIdHUq/65-as-an-admin-i-want-to-track-number-of-sign-up-clicks-on-all-pitch-pages-so-that-i-can-see-how-many-users-click-sign-up-and-make-?menu=filter&filter=label:orange,label:Onboarding,label:User,label:purple,@joekent3

### CODE REVIEW

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js [hunk 1]
```diff
@@ -40,6 +40,14 @@ function subscribeEvents() {
     const label = args !== null ? '#' + args.attr('id') : '';
     ga('send', 'event', 'Modal', 'Close', label);
   });
+
+  // Signup button on pitch page
+  const urlParts = window.location.pathname.split('/');
+  if (urlParts[urlParts.length - 1] === 'pitch') {
```
Comment (DFurnes) [inline lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Have you tested this with a normal account? As far as I  know, `/pitch` is just an override URL for admins.

For example, this is the pitch page URL for Lose Your V-Card (for a logged-out or not signed up user):

[`https://www.dosomething.org/us/campaigns/lose-your-v-card`](https://www.dosomething.org/us/campaigns/lose-your-v-card)

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js [hunk 2]
```diff
@@ -40,6 +40,14 @@ function subscribeEvents() {
...
+    $('#dosomething-signup-form-primary').on('submit', () => {
```
Comment (DFurnes) [inline lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Is this the ID for both the logged-in "sign up" button form and the logged-out "register and sign up" form? If not, will this still give us the results we're expecting for this test?

**Review by DFurnes — CHANGES_REQUESTED** [review]

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js (hunk 1 above)
Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Hmm thats a good point, maybe I could go by classname?

`class="campaign campaign--pitch pitch"`

Comment (DFurnes) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: That seems like a better idea. :thumbsup:

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js (hunk 2 above)
Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Good catch! `link--campaign-signup-login` is the other ID.

Potentially better option, listen for the button submit under `.header__signup`

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js [hunk 3]
```diff
@@ -40,6 +40,15 @@ function subscribeEvents() {
     const label = args !== null ? '#' + args.attr('id') : '';
     ga('send', 'event', 'Modal', 'Close', label);
   });
+
+  // Signup button on pitch page
+  const isPitchPage = $('.pitch').length > 0;
+  if (isPitchPage) {
+    // This applies to both authenticated & anonymous users. It also covers both modal register & login.
+    $('.header__signup form, #user-login-form, #user-register-form').on('submit', () => {
```
Comment (DFurnes) [inline lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Have you tested that `#user-register-form` doesn't trigger this analytics call if there are form validation errors? That may have to be a second event listener on the `Validation:Submitted` event so we don't get false positives from that form (see example usage above, `args` is the form ID).

**Review by DFurnes — CHANGES_REQUESTED** [review]

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js (hunk 3 above)
Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Looks like it doesn't trigger this event listener, the only GA event I see firing is for the validation error

Comment (DFurnes) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: So all working as expected (including actually submitting on successful validation)? If so, merge away!

**Review by DFurnes — APPROVED** [review]

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js (hunk 3 above)
Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: @DFurnes Actually looks like I didn't catch it the first time, I think I'll need to do the Validation Submitted approach

**Comment by itsjoekent** [comment]
@DFurnes okie, _now_ it works  with 
- new accounts
- logging in
- currently logged in

**Comment by DFurnes** [comment]
🚢

**Pull request merged**